lib/report_portal/patches/fariday.rb

```diff
@@ -0,0 +1,25 @@
+module Faraday
+  class Requests
+    # Middleware for supporting multi-part requests.
+    class Multipart
+      def create_multipart(env, params)
+        boundary = env.request.boundary
+        parts = process_params(params) do |key, value|
+          if begin
```

Comment:
This monkeypatch may have implications on other areas of code. Please discuss the proper way to address this use case with Faraday team.

Reply:
maybe report portal should have more restful API? where binary can be part of the JSON post and not separate multipart?

Reply:
Feel free to propose it to people responsible for the server side

Reply:
and it does

Reply:
https://github.com/reportportal/service-api/issues/822 not sure if this is the project for the issue

Reply:
Encoding files in base64 increases their size.

Reply:
https://github.com/lostisland/faraday/issues/769

Reply:
I see why
> Main reason for this is that mixing JSON and multipart is not really RESTful,

Reply:
I don't think it's relative, to this PR. 
it's not considered restful because the same resource may have completely different payloads:
1. JSON for a regular log entry
2. multipart for log with image

Reply:
It looks to me that this monkeypatch *might* break some code if that code happens to use a part similar to JSON but not actually JSON.
So it's better to use some other gem that supports sending such request without monkeypatch.
I created https://github.com/reportportal/agent-ruby/pull/60 to use HTTP gem.

Reply:
you should be concerned about reusing connection instead, as it impacts RP API service load and affecting performance.

Reply:
https://github.com/reportportal/agent-ruby/blob/efc07c0d0b5cfe8db93a7387a0a8778f45e3e776/lib/reportportal.rb#L210 - faraday supports different adapters

Reply:
> you should be concerned about reusing connection instead, as it impacts RP API service load and affecting performance.
I'm concerned about it as well. I plan to add ReportPortal::Client that holds access to connection and is used by clients.

Reply:
> faraday supports different adapters
That's true but as it's a wrapper, it provides only some features that underlying libraries support.